core/src/test/scala/org/apache/spark/CheckpointSuite.scala

```diff
@@ -238,6 +241,42 @@ trait RDDCheckpointTester { self: SparkFunSuite =>
   protected def generateFatPairRDD(): RDD[(Int, Int)] = {
     new FatPairRDD(sparkContext.makeRDD(1 to 100, 4), partitioner).mapValues(x => x)
   }
+
+  protected def testBasicCheckpoint(sc: SparkContext, reliableCheckpoint: Boolean): Unit = {
```

Comment:
nit: does this one test any special logic? If it's covered by other tests, not need to add it to increase the test time.